From 6604cce38fed748e98d3bd2bf9d0f368d67eeb3c Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 29 Jul 2015 23:13:33 +0200
Subject: [PATCH] kodi-config.cmake: use CMAKE_FIND_ROOT_PATH to fix
 cross-compilation

When cross-compiling, the location at build time of the libraries is
not the same as the one at run-time. The CMAKE_FIND_ROOT_PATH variable
is here to handle this difference, so use it in kodi-config.cmake.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
(rebased and simplified for Kodi 17.0-Krypton)
Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 project/cmake/KodiConfig.cmake.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/project/cmake/KodiConfig.cmake.in b/project/cmake/KodiConfig.cmake.in
index 76626ec..ffb8e1d 100644
--- a/cmake/KodiConfig.cmake.in
+++ b/cmake/KodiConfig.cmake.in
@@ -8,7 +8,7 @@ if(NOT @APP_NAME_UC@_PREFIX)
   set(@APP_NAME_UC@_PREFIX @APP_PREFIX@)
 endif()
 if(NOT @APP_NAME_UC@_INCLUDE_DIR)
-  set(@APP_NAME_UC@_INCLUDE_DIR @APP_INCLUDE_DIR@)
+  set(@APP_NAME_UC@_INCLUDE_DIR ${CMAKE_FIND_ROOT_PATH}/@APP_INCLUDE_DIR@)
 endif()
 if(NOT @APP_NAME_UC@_LIB_DIR)
   set(@APP_NAME_UC@_LIB_DIR @APP_LIB_DIR@)
@@ -20,7 +20,7 @@ endif()
 if(NOT WIN32)
   set(CMAKE_CXX_FLAGS "$ENV{CXXFLAGS} @CXX11_SWITCH@")
 endif()
-list(APPEND CMAKE_MODULE_PATH @APP_LIB_DIR@ @APP_DATA_DIR@/cmake)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_FIND_ROOT_PATH}/@APP_LIB_DIR@ ${CMAKE_FIND_ROOT_PATH}/@APP_DATA_DIR@/cmake)
 
 string(REPLACE ";" " " ARCH_DEFINES "@ARCH_DEFINES@")
 add_definitions(${ARCH_DEFINES} -DBUILD_KODI_ADDON)
-- 
2.5.0

